sources/packages/api/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -93,4 +99,69 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
       dayjs(endDate).diff(startDate, "days") < MAX_MFSAA_VALID_DAYS
     );
   }
+
+  /**
+   * Fetches the MSFAA number records which are not sent for request.
+   * This can be retrived by checking for date_requested column as null
+   * in the MSFAANumber table.
+   * @returns the records of the MSFAANumber table.
+   */
+  async getPendingMSFAARequest(
+    offeringIntensity: string,
+  ): Promise<MSFAANumber[]> {
+    return this.repo
+      .createQueryBuilder("msfaaNumber")
+      .select([
+        "msfaaNumber.id",
+        "students.id",
+        "referenceApplication.id",
+        "msfaaNumber.msfaaNumber",
+        "students.sin",
+        "institutionLocation.institutionCode",
+        "students.birthdate",
+        //"students.maritalStatus",
+        "users.lastName",
+        "users.firstName",
+        "students.gender",
+        "students.contactInfo",
+        "users.email",
+        "offering.offeringIntensity",
+      ])
+      .innerJoin("msfaaNumber.student", "students")
+      .innerJoin("students.user", "users")
+      .innerJoin("msfaaNumber.referenceApplication", "referenceApplication")
+      .innerJoin("referenceApplication.offering", "offering")
+      .innerJoin("offering.institutionLocation", "institutionLocation")
+      .where("msfaaNumber.dateRequested is null")
+      .andWhere("offering.offeringIntensity = :offeringIntensity", {
+        offeringIntensity,
+      })
+      .getMany();
+  }
+
+  /**
+   * Once the MSFAA request file is created, updates the
+   * date that the file was uploaded.
+   * @param msfaarequestIds records that are part of the generated
+   * file that must have the file sent name and date updated.
+   * @param dateSent date that the file was uploaded.
+   * @param [externalRepo] when provided, it is used instead of the
+   * local repository (this.repo). Useful when the command must be executed,
+   * for instance, as part of an existing transaction manage externally to this
+   * service.
+   * @returns the result of the update.
+   */
+  async updateRecordsInSentFile(
+    msfaarequestIds: number[],
+    dateRequested: Date,
+    externalRepo?: Repository<MSFAANumber>,
+  ) {
+    if (!dateRequested) {
+      throw new Error(
+        "Date sent field is not provided to update the MSFAA records.",
+      );
+    }
+    const repository = externalRepo ?? this.repo;
```

Comment:
Just to know, why are we using dynamic repo here?

Reply:
Added in the comments, 
   * @param [externalRepo] when provided, it is used instead of the
   * local repository (this.repo). Useful when the command must be executed,
   * for instance, as part of an existing transaction manage externally to this
   * service.

Reply:
Sorry @guru-aot I couldn't understand from the comments still. That is why asked.